doc/user/content/sql/tail.md

```diff
@@ -66,33 +136,67 @@ Syntax has changed. `WITH SNAPSHOT` is now `WITH (SNAPSHOT)`.
 {{</ version-changed >}}
 
 {{< version-changed v0.5.2 >}}
-The `TIMESTAMPS` option has been added.
+The [`PROGRESS`](#progress) option has been added.
 {{</ version-changed >}}
 
-### AS OF
+### `AS OF`
+
+The `AS OF` clauses specifies a point in time to start reporting all events for
+a given `TAIL` operation. If you don't use `AS OF`, Materialize will pick a
+timestamp automatically according to the following rules:
 
-`AS OF` is the specific point in time to start reporting all events for a given `TAIL`. If you don't
-use `AS OF`, Materialize will pick a timestamp itself.
+  - If the tailed relation is materialized, pick the latest time for which
+    results are computed.
+  - If the tailed relation is unmaterialized, pick time `0`.
```

Comment:
That the contents of the relation will be produced as tailed updates rather than as part of the snapshot. I think the `AS OF` and `SNAPSHOT` sections need to be merged to explain all this properly, which I'm out of time for. Blugh!